Fixes/code refactoring

VishnuKrishnathu opened the pull request.

VishnuKrishnathu committed:
Add environment configuration, jsconfig, and update storage service

- Introduced a new .env file for local proxy configuration.
- Added jsconfig.json for improved module resolution in the project.
- Updated package-lock.json with dependency version changes and licenses.
- Refactored storage service to utilize user preference slices for better state management.
- Modified hooks and components to integrate new storage methods and improve language handling.

VishnuKrishnathu committed:
Refactor storage service integration in ShikshalokamChat component

- Updated storage service methods to utilize user preference slices for improved state management.
- Changed data retrieval and storage functions to align with the new structure, enhancing user preference handling.
- Ensured consistent access to user-related data such as device ID, company, and session information across the component.

VishnuKrishnathu committed:
Refactor chat storage management and integrate user preference slices

- Updated the useSmartChatStorage hook to utilize user preference slices for managing chat history, enhancing state consistency.
- Refactored ShikshalokamVoiceBasedChat component to leverage new storage methods, improving data retrieval for user preferences and chat history.
- Introduced new chat data slices for better state management, ensuring a more organized approach to handling chat-related data.
- Enhanced storage service functions to support the new structure, facilitating improved user experience and data handling.

VishnuKrishnathu committed:
Refactor chat storage and websocket configuration

- Updated the useSmartChatStorage hook to manage chat history more effectively using local and session storage.
- Adjusted websocket protocol handling in the ShikshalokamVoiceBasedChat component for improved connection reliability.
- Cleaned up commented code and ensured consistent chat history updates across user interactions.
- Removed unused imports to streamline the codebase.

VishnuKrishnathu committed:
Update environment configuration and refactor user preference handling

- Set local proxy and websocket host in the .env file for development.
- Refactored i18n.js to utilize site data slices for language management.
- Updated GuestPage, Login, and ShikshalokamChat components to replace axios calls with new API methods for login and location retrieval.
- Cleaned up unused imports and improved state management in user data slices.
- Enhanced chat session handling by integrating new API endpoints for better data retrieval.

VishnuKrishnathu committed:
commented out

themohitbharti committed:
Refactor API service methods: Uncomment postWithoutAuth and getWithoutAuth functions; Comment out submitFeedBack and savePTMQuestion functions for future use.

themohitbharti committed:
Refactor API calls in UnifiedChat, ShikshalokamChat, and SsoFlow components

- Updated UnifiedChat to use getSessionDetailsApi instead of getSessionDetails.
- Refactored ShikshalokamChat to replace getSessionDetails with getSessionDetailsApi and getIpLocation with getIpLocationApi.
- Modified SsoFlow to utilize getSessionDetailsApi for session management.
- Cleaned up imports and improved code readability across the affected files.

themohitbharti committed:
Add project API endpoints and refactor user profile handling in components

themohitbharti committed:
Add AI4Bharat API endpoints for audio processing and transliteration; refactor related service calls

themohitbharti committed:
Refactor story API service and update imports in UnifiedVoiceBasedChat and PdfDownloader components

- Moved API functions from `src/pages/story/api.service.js` to `src/api/endpoints.js` and updated import paths accordingly.
- Removed the now obsolete `api.service.js` file.
- Adjusted import statements in `UnifiedVoiceBasedChat.jsx` and `pdfDownloader.js` to reflect the new API structure.
- Enhanced the PdfDownloader component by improving the handling of story media and PDF generation logic.

VishnuKrishnathu committed:
Refactor language handling and state management across components

- Updated i18n.js to utilize the new getChatLanguage method for improved language retrieval.
- Refactored FlowSelection, Header, LanguageSelector, and ShowPageButton components to remove direct userLanguage props and replace them with state management from siteDataLocalStore.
- Enhanced useFlow and useLanguage hooks to streamline language processing and flow selection.
- Introduced useStorage hook for better state management in FlowSelection and FormData components.
- Cleaned up commented code and improved overall code organization for better maintainability.

VishnuKrishnathu committed:
Refactor session management and language handling across components

- Introduced SESSION_USECASE_TYPE constant for improved session use case management.
- Updated CommonHomePage, LanguageSelectionGrid, and useFlow hooks to utilize new session constants.
- Streamlined language selection logic in LanguageSelectionGrid and useLanguage hooks.
- Enhanced state management by integrating useStorage for chat language and selection status.
- Cleaned up unused imports and improved overall code organization for better maintainability.

VishnuKrishnathu committed:
Add .vscode/ to .gitignore for improved development environment management

themohitbharti committed:
Add getProfileUserApi function and update PdfDownloader to fetch user profile data

themohitbharti committed:
Add upload API endpoints and integrate S3 upload functionality

themohitbharti committed:
deleted services.js

VishnuKrishnathu committed:
Merge branch 'fixes/code-refactoring' of https://github.com/VishnuKrishnathu/mohini-app-frontend into fixes/api-structuring

VishnuKrishnathu committed:
Merge pull request #2 from VishnuKrishnathu/fixes/api-structuring

Fixes/api structuring

VishnuKrishnathu committed:
Refactor language and chat data management across components

- Updated useLanguage and useSmartChatStorage hooks to utilize useStorage for improved state management.
- Refactored ShikshalokamChat and UnifiedVoiceBasedChat components to streamline user data handling and session management.
- Enhanced siteData and userData slices to include new state properties and methods for better data handling.
- Cleaned up unused imports and improved overall code organization for better maintainability.

VishnuKrishnathu committed:
Enhance chat functionality and state management

- Added `getCompanyBotApi` to fetch company-specific bot data from the API.
- Updated `urls.js` to include the new endpoint for company bot retrieval.
- Refactored `FlowSelection`, `ShowPageButton`, and `useAudio` components to remove unused `controllerRef` prop for cleaner code.
- Introduced `useConfirmationPopup` hook for managing confirmation dialogs across the application.
- Improved state management in `shikshalokamChat` by integrating access token handling and optimizing user data retrieval.
- Updated `.gitignore` to include `.env` for better environment management.

VishnuKrishnathu committed:
Refactor chat components and enhance state management

- Added logging in `ShikshalokamChat` to monitor `companyName` and `isLoading` state changes.
- Updated dependencies in `ShikshalokamVoiceBasedChat` to improve error handling and session management.
- Cleaned up unused code and improved state handling in `voice-chat.js`.
- Refactored storage service to utilize a mapping for local and session storage slices, enhancing modularity and maintainability.

VishnuKrishnathu committed:
Refactor chat state management and improve code consistency

- Updated `App.js` to standardize route object formatting for better readability.
- Refactored `voice-chat.js` to enhance state management, including the addition of reconnect attempts and improved WebSocket handling.
- Renamed `intro_message` to `introMessage` in state management for consistency across the application.
- Cleaned up unused code and improved overall organization in `chatData` and `userData` slices for better maintainability.

VishnuKrishnathu committed:
Add ReportEditor component and useChatWebhook hook for enhanced chat functionality

- Introduced `ReportEditor` component for editing reports within the chat interface.
- Implemented `useChatWebhook` hook to manage WebSocket connections, including automatic reconnection logic.
- Updated `voice-chat.js` to integrate the new `ReportEditor` and `useChatWebhook`, improving chat state management and user experience.
- Refactored imports and organized code for better maintainability and readability.

VishnuKrishnathu committed:
Enhance configuration and code organization

- Updated `.env` file to include new environment variables for WebSocket and S3 upload configurations.
- Added `.prettierrc` for consistent code formatting across the project.
- Introduced `.vscode/settings.json` for improved development environment settings.
- Refactored `App.js` to streamline route definitions and improve readability.
- Cleaned up imports and standardized code formatting in various components for better maintainability.

VishnuKrishnathu committed:
route handling

VishnuKrishnathu committed:
Refactor chat API endpoints and enhance functionality

- Updated `chat.js` to improve API endpoint structure and error handling.
- Introduced `getStoryBySessionAPI` for fetching story data based on session ID.
- Cleaned up import statements for consistency across files.
- Refactored `UnifiedVoiceBasedChat` and `DownloadStoryButton` to utilize the new `getStoryBySessionAPI` function, improving data retrieval and state management.
- Enhanced overall code readability and maintainability by standardizing formatting and removing unused code.

VishnuKrishnathu committed:
Refactor story API endpoints and improve error handling

- Simplified the `createAuthRequest` function by consolidating parameters and enhancing error messages.
- Updated `getStoryById`, `getStoryAllMedia`, `createStoryMedia`, and `partialUpdateStoryById` functions to include better error handling and validation checks.
- Refactored `UnifiedChat` component to utilize new state management methods and improve readability.
- Cleaned up unused code and standardized import statements across files for better maintainability.

VishnuKrishnathu committed:
Merge branch 'ELEVATE-Project:release-1.0.4' into release-1.0.4

VishnuKrishnathu committed:
Refactor story media update logic and API endpoint paths

- Simplified the `updateStoryMedia` function by removing unnecessary parameters and utilizing a more dynamic approach for appending form data.
- Updated API endpoint paths in `urls.js` for better organization and clarity.
- Refactored `voice-chat.js` and `upload-images/index.js` to integrate the new `updateStoryMedia` function, enhancing media update functionality.
- Cleaned up unused code and improved overall readability across multiple components.

apsrcreatix commented on src/App.js:
```diff
@@ -1,91 +1,77 @@
-  { path: ROUTES.SHIKSHALOKAM_GUEST_LISTENING_CHAT, element: <ShikshalokamChat type={sessionFlowName.ListeningActivity} />},
-  { path: ROUTES.SHIKSHALOKAM_GUEST_MI_STORY, element: <ShikshalokamChat type={sessionFlowName.GuestMiStory} />},
-  
+  { path: ROUTES.SHIKSHALOKAM_VOICE_CHAT_LOGIN, element: <Shikshalokam type={"shikshalokam"} variant={"publicBot"} /> },
```
Will there be more types? if yes, keep it inside object not directly hardcoded

apsrcreatix commented on src/App.js:
```diff
@@ -1,91 +1,77 @@
-  { path: ROUTES.SHIKSHALOKAM_GUEST_MI_STORY, element: <ShikshalokamChat type={sessionFlowName.GuestMiStory} />},
-  
+  { path: ROUTES.SHIKSHALOKAM_VOICE_CHAT_LOGIN, element: <Shikshalokam type={"shikshalokam"} variant={"publicBot"} /> },
+  { path: ROUTES.SHIKSHALOKAM_VOICE_CHAT, element: <ShikshalokamVoiceBasedChat type={"shikshalokam"} variant={"publicBot"} /> },
```
remove if we are not using this --> "public bot", route is no longer in use, keep the component, only remove the url, confirm with nivedita

apsrcreatix commented on src/store/slices/chatData/state.js:
```diff
@@ -0,0 +1,71 @@
+export const INITIAL_STATE = (set, get, store) => ({
+  showFileInput: null,
```
@KUNALTEMPEST  can you verify if any of these can be removed or are redundant

apsrcreatix commented on src/store/slices/persistent/userPreference.js:
```diff
@@ -0,0 +1,127 @@
+import { create } from 'zustand'
```
will be deleted in upcoming commmits

apsrcreatix commented on src/store/slices/session/userPreference.js:
```diff
@@ -0,0 +1,128 @@
+import { create } from 'zustand'
```
will be deleted in upcomming commits

apsrcreatix commented on src/pages/shikshalokamChat.js:
```diff
@@ -1,192 +1,168 @@
+  const navigate = useNavigate()
+  const [isLoading, setIsLoading] = useState(false)
+
+  const { setFirstName } = useStorage(STORE_NAME_CONSTANTS.USER_DATA).getState()
```
useSiteDataStorage, useChatDataStorage, useUserDataStorage ---> these can be implemented to avoid redundant code and it helps improve code maintainability.

apsrcreatix commented on src/services/api.service.js:
```diff
@@ -158,21 +158,21 @@ export async function transliterateApi(
-// 	sent_at: "2025-07-04T12:00:00Z",
-// 	audio_url: null
-// };
+ const squestion = {
```
this was for sample

apsrcreatix commented on src/pages/ssoFlow.jsx:
```diff
@@ -1,122 +1,116 @@
-      const sessionId = urlParams.get("sessionId");
-      const languagePassed = urlParams.get("language");
-      let rerouteRaw = urlParams.get("rerouteUrl") || "";
+    setLanguage("en")
```
change url param string to come from object, same for language as well

apsrcreatix commented on src/pages/ssoFlow.jsx:
```diff
@@ -1,122 +1,116 @@
-            setInStorage('isNewChatOpen', JSON.stringify(true), flow_type, localStorage);
-            setInStorage('projectId', JSON.stringify(projectId), flow_type, localStorage);
-            setInStorage('taskId', JSON.stringify(taskId), flow_type, localStorage);
+            setInStorage("ssoRerouteURL", rerouteUrl, flow_type, localStorage)
```
setinstorage, getfromstroage needs to be removed

apsrcreatix commented on src/pages/ShikshalokamMegaPTM/PTMVoiceBasedChat.js:
```diff
@@ -1,216 +1,181 @@
+  const audioRef = useRef()
+  const textAreaRef = useRef(null)
+  const lastBotMessageIndex = useRef(-1)
+  const FLOW_ROUTE = "/mega_ptm"
```
change it with enums

apsrcreatix commented on src/components/Login.jsx:
```diff
@@ -58,23 +60,22 @@ function Login({ type, variant }) {
   const handleLanguageChange = (e) => {
     setUserLanguage(e.target.value);
-    localStorage.setItem('preferred_route', JSON.stringify(e?.target?.value));
+    localStorage.setItem("preferred_route", JSON.stringify(e?.target?.value));
```
do not directly set like this, use hook or something

apsrcreatix commented on src/components/GuestPage.jsx:
```diff
@@ -46,94 +50,99 @@ function GuestPage() {
-        }
-      }
-      
+          city: localStorage.getItem("ip_city") || "",
```
hook it instead of directly using

VishnuKrishnathu commented on src/App.js:
```diff
@@ -1,91 +1,77 @@
+const protected_routes = []
 
 const unprotected_old_routes = [
-  { path: ROUTES.SHIKSHALOKAM_VOICE_CHAT_LOGIN, element: <Shikshalokam type={'shikshalokam'} variant={'publicBot'}/> },
```
Needs confirmation from Niveditha

VishnuKrishnathu committed:
Enhance routing and storage management

- Added new routes for guest voice chat and listening chat in `url.js`.
- Updated `App.js` to include the new guest voice chat route.
- Refactored `useStorage` hook in `useStorage.js` to create specific hooks for chat, user, and site data storage.
- Modified `voice-chat.js` and `StoryActionsModule.js` to utilize the new storage hooks for improved state management.
- Cleaned up commented code and improved overall readability across multiple components.

themohitbharti committed:
Refactor storage hooks to use appropriate storage methods across components

themohitbharti committed:
Refactor storage usage to utilize useChatStorage across components

themohitbharti committed:
created separate hooks instead of useStorage

VishnuKrishnathu committed:
Refactor API endpoint imports and improve code consistency

- Updated import statements across multiple API endpoint files to use named imports for `API_ENDPOINTS` instead of default imports.
- Refactored various API functions to enhance readability by removing unnecessary line breaks and standardizing formatting.
- Cleaned up unused code and improved overall consistency in the codebase, particularly in the `LanguageSelector` and `SsoFlow` components.
- Removed the deprecated `useNavigation` hook to streamline the codebase.

VishnuKrishnathu committed:
reverted back websocket connection

VishnuKrishnathu committed:
Merge branch 'fixes/code-refactoring' of https://github.com/VishnuKrishnathu/mohini-app-frontend into fixes/storage-hooks

VishnuKrishnathu committed:
Refactor storage hook usage for consistency across components

- Updated the usage of `useSiteDataLocalStore` to streamline state management in `LanguageSelectionGrid`, `useLanguage`, `commonPage`, `UnifiedChat`, and `UnifiedVoiceBasedChat`.
- Improved readability by removing unnecessary parentheses in hook calls.
- Adjusted the websocket connection logic in `voice-chat.js` for clarity and consistency.

VishnuKrishnathu committed:
Merge branch 'release-1.0.4' of https://github.com/ELEVATE-Project/mohini-app-frontend into fixes/code-refactoring

VishnuKrishnathu committed:
changes

VishnuKrishnathu committed:
changes

VishnuKrishnathu committed:
changes

VishnuKrishnathu committed:
Merge branch 'release-1.0.4' of https://github.com/ELEVATE-Project/mohini-app-frontend into fixes/storage-hooks

VishnuKrishnathu committed:
Merge branch 'fixes/storage-hooks' of https://github.com/VishnuKrishnathu/mohini-app-frontend into fixes/code-refactoring

VishnuKrishnathu committed:
Refactor SsoFlow component for improved clarity and consistency

- Removed unnecessary parameter from clearFromStorage function call.
- Adjusted the order of state management hooks for better organization.

VishnuKrishnathu committed:
Update websocket connection logic and comment out accessToken check in ShikshalokamChat

- Reverted the websocket protocol assignment to use dynamic protocol based on the current window location in voice-chat.js.
- Commented out the accessToken check in ShikshalokamChat to allow for further setup processing.
